src/components/hw-design/hw-button.tsx

```diff
@@ -0,0 +1,38 @@
+import { ReactNode } from "react";
+
+/**
+ * A reusable Hack Western button component with variants and customizable styles and attributes.
+ * To override any of the default tailwind styles, pass in custom styles with the className prop, use ! before a className to force style override
+ * @param props - The props for the Button component.
+ * @param props.className - Additional CSS classes to apply to the button.
+ */
+function HWButton({
+  text,
+  className = "",
+  variant,
+  children,
+  ...props
+}: {
+  text: string;
+  className?: string;
+  children?: ReactNode;
+  variant: "primary" | "destructive" | "outline" | "subtle";
```

Comment:
type is wrong here, it doesn't include all other button attributes (like `onClick` for example) which will result in type errors when it's actually used. I think we should avoid figuring this out, and just restyle the shadcn components instead.

Reply:
oh this is just a one line fix, I can set that up rn

Reply:
ok used the shadcn as a base for these components as well, but im gonna keep the components cause they already have all the styles and stuff done on them lol

Reply:
ok but like we can just change the styles directly on the button component though. We shouldn't be scared to replace/change the shadcn styles, the whole point of them being in our codebase is so that we can replace them. Just want to stray away from having two base Button components, especially when we can change/add style variants to one

Reply:
Also, when we add more shadcn components, by default it uses any dependent components from the ui folder. So if we wanted them to use our default styling, we'd have to go about changing the imports to import this button.